watsononlinestore/watson_online_store.py

```diff
@@ -267,6 +267,19 @@ def get_discovery_response(self, input_text):
             # This dumps a ton of results for us to peruse:
             pprint(discovery_response)
 
+        filtered_results = []
```

Comment:
num_matches is used to correctly populate discovery_response['matching_results']. This change maintains the structure of the response object returned by Watson Discovery. Even though we may not use this dict value now, we may in the future. In fact, it's probably cleaner to query this value in format_discovery_response instead of looking if 'results' is empty.